lib/BaseController.js

```diff
@@ -544,7 +544,17 @@ module.exports = class BaseController {
       let debug = objectPath.get(liveResource, ['metadata', 'labels', 'deploy.razee.io/debug']) ||
         objectPath.get(liveResource, ['metadata', 'labels', 'kapitan.razee.io/debug'], 'false');
       if (debug.toLowerCase() === 'true') {
+        this.log.warn(`${uri}: Debug enabled on resource, adding deploy.razee.io/pending-configuration`);
+        if (objectPath.get(file, ['metadata', 'annotations']) === null) {
+          objectPath.set(file, ['metadata', 'annotations'], {});
+        }
+        await this.patchSelf({ metadata: { annotations: { 'deploy.razee.io/pending-configuration': JSON.stringify(file) } } });
         return { statusCode: 200, body: liveResource };
+      } else {
+        let pendingApply = objectPath.get(liveResource, ['metadata', 'annotations', 'deploy.razee.io/pending-configuration']);
+        if (pendingApply) {
+          objectPath.set(file, ['metadata', 'annotations', 'deploy.razee.io/pending-configuration'], null);
+        }
```

Comment:
in order for the annotation to end up on the resource that is pending update, i think this is what you want
```suggestion
        this.log.warn(`${uri}: Debug enabled on resource: skipping modifying resource - adding annotation deploy.razee.io/pending-configuration.`);
        let patchObject = { metadata: { annotations: { 'deploy.razee.io/pending-configuration': JSON.stringify(file) } } };
        let res = await krm.mergePatch(name, namespace, patchObject);
        return { statusCode: 200, body: res };
      } else {
        let pendingApply = objectPath.get(liveResource, ['metadata', 'annotations', 'deploy.razee.io/pending-configuration']);
        if (pendingApply) {
          objectPath.set(file, ['metadata', 'annotations', 'deploy.razee.io/pending-configuration'], null);
        }
```